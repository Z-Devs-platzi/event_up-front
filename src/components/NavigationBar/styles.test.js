import React from 'react';
import {
  NavigationContainer,
  NavigationHeader,
  NavigationMenu,
} from './styles';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Test NavigationBar styled-components', () => {
  it('should test the styled-component NavigationContainer', () => {
    const containerStyles = renderer.create(<NavigationContainer />).toJSON();
    expect(containerStyles).toMatchSnapshot();
    expect(containerStyles).toHaveStyleRule('position', 'fixed');
  });
  it('should test the styled-component NavigationHeader', () => {
    const headerStyles = renderer.create(<NavigationHeader />).toJSON();
    expect(headerStyles).toMatchSnapshot();
    expect(headerStyles).toHaveStyleRule(
      'background-color',
      'var(--primary-color)'
    );
  });
  it('should test the styled-component NavigationMenu', () => {
    const menuStyles = renderer.create(<NavigationMenu />).toJSON();
    expect(menuStyles).toMatchSnapshot();
    expect(menuStyles).toHaveStyleRule('background', 'var(--gradient-green)');
  });
});
