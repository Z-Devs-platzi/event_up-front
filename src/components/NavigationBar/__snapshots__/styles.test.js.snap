// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavigationBar styled-components should test the styled-component NavigationContainer 1`] = `
.c0 {
  position: fixed;
  z-index: 100;
  height: 100vh;
  width: 248px;
}

@media screen and (max-width:900px) {
  .c0 {
    position: relative;
  }
}

<div
  className="c0"
/>
`;

exports[`Test NavigationBar styled-components should test the styled-component NavigationHeader 1`] = `
.c0 {
  padding: 44px 50px 0;
  background-color: var(--primary-color);
}

.c0 .logo {
  width: 148px;
}

@media screen and (max-width:900px) {
  .c0 {
    width: 100vw;
    height: 56px;
    padding: 16px;
  }

  .c0 .logo {
    display: block;
    margin: 0 auto;
  }
}

<header
  className="c0"
/>
`;

exports[`Test NavigationBar styled-components should test the styled-component NavigationMenu 1`] = `
.c0 {
  position: relative;
  height: calc(100vh - 80px);
  padding: 38px 32px 0;
  background: var(--gradient-green);
}

@media screen and (max-width:900px) {
  .c0 {
    position: fixed;
    bottom: 0;
    left: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: 100vw;
    height: 60px;
    padding: 0 10px;
    background: var(--primary-color);
  }
}

<ul
  className="c0"
/>
`;
